cmd/kubeadm/app/phases/upgrade/compute_test.go

```diff
@@ -221,7 +182,45 @@ func TestGetAvailableUpgrades(t *testing.T) {
 			},
 			allowExperimental: false,
 			errExpected:       false,
-			etcdClient:        etcdClient,
+		},
+		{
+			name: "simple patch version upgrade with external etcd",
+			vg: &fakeVersionGetter{
+				clusterVersion: v1Y1.String(),
+				kubeletVersion: v1Y1.String(), // the kubelet are on the same version as the control plane
+				kubeadmVersion: v1Y2.String(),
+
+				stablePatchVersion: v1Y3.String(),
+				stableVersion:      v1Y3.String(),
+			},
+			beforeDNSType:    kubeadmapi.CoreDNS,
+			beforeDNSVersion: fakeCurrentCoreDNSVersion,
+			dnsType:          kubeadmapi.CoreDNS,
+			externalEtcd:     true,
+			expectedUpgrades: []Upgrade{
+				{
+					Description: fmt.Sprintf("version in the v%d.%d series", v1Y0.Major(), v1Y0.Minor()),
+					Before: ClusterState{
+						KubeVersion: v1Y1.String(),
+						KubeletVersions: map[string]uint16{
+							v1Y1.String(): 1,
+						},
+						KubeadmVersion: v1Y2.String(),
+						DNSType:        kubeadmapi.CoreDNS,
+						DNSVersion:     fakeCurrentCoreDNSVersion,
+						EtcdVersion:    "",
+					},
+					After: ClusterState{
+						KubeVersion:    v1Y3.String(),
+						KubeadmVersion: v1Y3.String(),
+						DNSType:        kubeadmapi.CoreDNS,
+						DNSVersion:     constants.CoreDNSVersion,
+						EtcdVersion:    getEtcdVersion(v1Y3),
```

Comment:
maybe we should populate the `After` struct with an empty `EtcdVersion` on external etcd.